src/transfer.cpp

```diff
@@ -150,9 +150,10 @@ bool Transfer::serialize(string *d)
 
     d->append((const char*)&type, sizeof(type));
 
-    ll = (unsigned short)localfilename.editStringDirect()->size();
+    const auto& tmpstr = localfilename.platformEncoded();
```

Comment:
unimportant: RVO will probably make `const auto tmpstr = ... ` equal to this

Reply:
For linux, it is a reference to the string stored inside the LocalPath.   But for WIN32, it's a reference to the temporary string returned.